app/reducers/cart-reducer.jsx

```diff
@@ -0,0 +1,40 @@
+import {
+  RECEIVE_CART_ITEM,
+  UPDATE_CART_ITEM,
+  REMOVE_CART_ITEM
+} from '../constants';
+
+
+const initialCartState = {
+    list: {}
+};
+
+export default function (state = initialCartState, action) {
+
+  const newState = Object.assign({}, state);
+
+  switch (action.type) {
+
+    //Add item to cart, which is an object whose keys are product id.
+    case RECEIVE_CART_ITEM:
+      newState.list[action.product.id.toString()] = action.product
+      break;
+
+    //Update the quantity on an item in the cart.
+    case UPDATE_CART_ITEM:
+      newState.list.action.product.id.toString().quantity = action.product.quantity
```

Comment:
If you do a toString() you'll convert the variable to a string, so you won't be able to do a .quantity off this because it's not an object anymore. Looks like a bug... Did you mean to use it as a key?